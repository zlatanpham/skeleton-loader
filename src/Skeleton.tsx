import * as React from 'react';
import styled from 'styled-components';

const Container = styled.div`
  background-color: red;
`;

export class Skeleton extends React.Component {
  render() {
    return <Container>Hello</Container>;
  }
}
