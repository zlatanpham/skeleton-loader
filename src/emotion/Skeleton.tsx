import * as React from 'react';
import { CSSObject, SerializedStyles } from '@emotion/core';
import styled from '@emotion/styled';
import {
  getStyles,
  defaultBaseColor,
  defaultHighlightColor,
  CSSProperty,
  bindProperty
} from './shared';
import {
  SkeletonThemeConsumer,
  SkeletonThemeProps
} from './SkeletonThemeProvider';

const Container = styled.div<
  DefaultSkeletonProps & {
    theme: SkeletonThemeProps;
    cssProps: DefaultSkeletonCSSProps;
  }
>`
  ${props => getStyles(props)}
  ${props => bindProperty('width', props.cssProps.width)};
  ${props => bindProperty('height', props.cssProps.height)};
  ${props => bindProperty('border-radius', props.cssProps.radius)};
  line-height: 1;
`;

export interface SkeletonCSSProps {
  width?: CSSProperty;
  height?: CSSProperty;
  radius?: CSSProperty;
}

export interface SkeletonProps extends React.HTMLAttributes<HTMLDivElement> {
  customStyle?: CSSObject | SerializedStyles;
  baseColor?: string;
  highlightColor?: string;
  duration?: number;
}

export interface DefaultSkeletonCSSProps {
  width: CSSProperty;
  height: CSSProperty;
  radius: CSSProperty;
  duration: number;
}

export interface DefaultSkeletonProps
  extends React.HTMLAttributes<HTMLDivElement> {
  duration: number;
  customStyle?: CSSObject | SerializedStyles;
  baseColor: string;
  highlightColor: string;
}

export class Skeleton extends React.Component<
  SkeletonProps & SkeletonCSSProps
> {
  static defaultProps: DefaultSkeletonProps & DefaultSkeletonCSSProps = {
    width: '100%',
    radius: '0px',
    height: '20px',
    duration: 1.5,
    baseColor: defaultBaseColor,
    highlightColor: defaultHighlightColor
  };

  shouldComponentUpdate() {
    return false;
  }

  render() {
    const { duration, width, radius, height, ...rest } = this.props;
    const cssProps = {
      height,
      width,
      radius,
      duration
    } as DefaultSkeletonCSSProps;

    return (
      <SkeletonThemeConsumer>
        {theme => (
          <Container
            cssProps={cssProps}
            theme={theme}
            {...rest as DefaultSkeletonProps}
          />
        )}
      </SkeletonThemeConsumer>
    );
  }
}
