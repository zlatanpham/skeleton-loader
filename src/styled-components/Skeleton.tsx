import * as React from 'react';
import styled, { CSSObject } from 'styled-components';
import {
  getStyles,
  defaultBaseColor,
  defaultHighlightColor,
  CSSProperty,
  defaultDuration,
  bindProperty
} from './shared';
import {
  SkeletonThemeConsumer,
  SkeletonThemeProps
} from './SkeletonThemeProvider';

const Container = styled.div<
  DefaultSkeletonProps & {
    theme: SkeletonThemeProps;
    cssProps: DefaultSkeletonCSSProps;
  }
>`
  ${props => getStyles(props)}
  ${props => bindProperty('width', props.cssProps.width)};
  ${props => bindProperty('height', props.cssProps.height)};
  ${props => bindProperty('border-radius', props.cssProps.radius)};
  line-height: 1;
`;

export interface SkeletonCSSProps {
  width?: CSSProperty;
  height?: CSSProperty;
  radius?: CSSProperty;
}

export interface SkeletonProps extends React.HTMLAttributes<HTMLDivElement> {
  customStyle?: CSSObject | TemplateStringsArray;
  baseColor?: string;
  highlightColor?: string;
  duration?: number;
}

export interface DefaultSkeletonCSSProps {
  width: CSSProperty;
  height: CSSProperty;
  radius: CSSProperty;
}

export interface DefaultSkeletonProps
  extends React.HTMLAttributes<HTMLDivElement> {
  duration: number;
  customStyle?: CSSObject | TemplateStringsArray;
  baseColor: string;
  highlightColor: string;
}

export class Skeleton extends React.Component<
  SkeletonProps & SkeletonCSSProps
> {
  static defaultProps: DefaultSkeletonProps & DefaultSkeletonCSSProps = {
    width: '100%',
    radius: '0px',
    height: '20px',
    duration: defaultDuration,
    baseColor: defaultBaseColor,
    highlightColor: defaultHighlightColor
  };

  shouldComponentUpdate() {
    return false;
  }

  render() {
    const { width, height, radius, ...rest } = this.props;
    const cssProps = {
      height,
      width,
      radius
    };
    return (
      <SkeletonThemeConsumer>
        {theme => (
          <Container
            cssProps={cssProps as DefaultSkeletonCSSProps}
            {...rest as DefaultSkeletonProps}
            theme={theme}
          />
        )}
      </SkeletonThemeConsumer>
    );
  }
}
