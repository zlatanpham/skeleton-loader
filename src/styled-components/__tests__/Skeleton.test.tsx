import 'jest-styled-components';
import * as React from 'react';
import { render } from '@testing-library/react';
import { Skeleton } from '../Skeleton';

describe('<Skeleton />', () => {
  it('Should match snapshot', () => {
    const { container } = render(<Skeleton width="200px" />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
