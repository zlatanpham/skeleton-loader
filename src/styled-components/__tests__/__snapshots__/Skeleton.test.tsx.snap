// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Skeleton /> Should match snapshot 1`] = `
.c0 {
  background-color: #eeeeee;
  background-image: linear-gradient( 90deg, #eeeeee, #f5f5f5, #eeeeee );
  background-size: 200px 100%;
  background-repeat: no-repeat;
  display: inline-block;
  -webkit-animation: ceFrVu 1.5s ease-in-out infinite;
  animation: ceFrVu 1.5s ease-in-out infinite;
  color: transparent !important;
  width: 200px;
  height: 20px;
  border-radius: 0px;
  line-height: 1;
}

.c0 > * {
  color: transparent !important;
}

<div
  class="c0"
/>
`;
